### PR TITLE
Device manager - extract device deletion logic into util

### PR DESCRIPTION
**Pull request by kerryarchibald**
PSG-660

Extracts device deletion logic into util and tests it, to be used in new device manager page.

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/components/views/settings/DevicesPanel-test.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+            toggleDeviceSelection(container, device2.device_id);
+
+            act(() => {
+                fireEvent.click(getByTestId('sign-out-devices-btn'));
+            });
+
+            await flushPromises();
+            // modal rendering has some weird sleeps
+            await sleep(10);
```
Comment (SimonBrandner) [inline test/components/views/settings/DevicesPanel-test.tsx:131]: This `ms`, right?

File: test/components/views/settings/DevicesPanel-test.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            await flushPromises();
+
+            // devices refreshed
+            expect(mockClient.getDevices).toHaveBeenCalled();
+            // and rerendered
+            expect(container.getElementsByClassName('mx_DevicesPanel_device').length).toEqual(2);
+        });
+
+        it('deletes selected devices when interactive auth is required', async () => {
+            mockClient.deleteMultipleDevices
```
Comment (SimonBrandner) [inline test/components/views/settings/DevicesPanel-test.tsx:105]: Should we also test this gets called correctly?

File: test/components/views/settings/DevicesPanel-test.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                // require auth
+                .mockRejectedValueOnce(interactiveAuthError)
+                // then succeed
+                .mockResolvedValueOnce({});
+
+            mockClient.getDevices
+                .mockResolvedValueOnce({ devices: [device1, device2, device3] })
+                // pretend it was really deleted on refresh
+                .mockResolvedValueOnce({ devices: [device1, device3] });
```
Comment (SimonBrandner) [inline test/components/views/settings/DevicesPanel-test.tsx:160]: Does this ever get called before we clear the mock,

**Review by SimonBrandner — APPROVED** [review]
Looking great

File: test/components/views/settings/DevicesPanel-test.tsx (hunk 3 above)
Comment (kerryarchibald) [reply test/components/views/settings/DevicesPanel-test.tsx:160]: clearing the mock clears the mock counter, not the mock implementations (that's `mockReset`)

Comment (SimonBrandner) [reply test/components/views/settings/DevicesPanel-test.tsx:160]: Ah, sorry!

File: test/components/views/settings/DevicesPanel-test.tsx (hunk 1 above)
Comment (SimonBrandner) [reply test/components/views/settings/DevicesPanel-test.tsx:131]: I think it can be useful to add the unit to a comment

**Review by SimonBrandner — APPROVED** [review]

**Pull request merged**